libcaja-private/caja-customization-data.c

```diff
@@ -459,45 +459,36 @@ format_name_for_display (CajaCustomizationData *data, const char* name)
 static void
 load_name_map_hash_table (CajaCustomizationData *data)
 {
-    char *xml_path;
-
     xmlDocPtr browser_data;
     xmlNodePtr category_node, current_node;
 
     /* allocate the hash table */
     data->name_map_hash = g_hash_table_new_full (g_str_hash, g_str_equal, g_free, g_free);
 
-    /* build the path name to the browser.xml file and load it */
-    xml_path = g_build_filename (CAJA_DATADIR, "browser.xml", NULL);
-    if (xml_path)
+    /* load the browser.xml file */
+    if ((browser_data = xmlParseFile (CAJA_DATADIR "/browser.xml")) != NULL)
     {
-        browser_data = xmlParseFile (xml_path);
-        g_free (xml_path);
+        /* get the category node */
+        category_node = eel_xml_get_root_child_by_name_and_property (browser_data, "category", "name", data->customization_name);
+        current_node = category_node->children;
 
-        if (browser_data)
+        /* loop through the entries, adding a mapping to the hash table */
+        while (current_node != NULL)
         {
-            /* get the category node */
-            category_node = eel_xml_get_root_child_by_name_and_property (browser_data, "category", "name", data->customization_name);
-            current_node = category_node->children;
+            char *filename, *display_name;
 
-            /* loop through the entries, adding a mapping to the hash table */
-            while (current_node != NULL)
+            display_name = xmlGetProp (current_node, "display_name");
```

Comment:
We get a build warning.
```
caja-customization-data.c:480:54: warning: passing 'char [13]' to parameter of type 'const xmlChar *' (aka 'const unsigned char *') converts between pointers to integer types with different sign [-Wpointer-sign]

            display_name = xmlGetProp (current_node, "display_name");

                                                     ^~~~~~~~~~~~~~
```
https://travis-ci.org/github/mate-desktop/caja/jobs/757376330#L3473

Reply:
this warning already exists on master branch, we only remove the `xml_path` variable in this PR.

Reply:
OK